### PR TITLE
[NTUSER] Implement NtUserUpdateInputContext

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Implementing Japanese input...
JIRA issue: [CORE-11700](https://jira.reactos.org/browse/CORE-11700)

## Proposed changes

- Modify `NtUserUpdateInputContext` prototype.
- Implement `NtUserUpdateInputContext` function.
- Add `enum UPDATE_INPUT_CONTEXT (UIC_*)` to `undocuser.h`.

### CODE REVIEW

File: win32ss/user/ntuser/ntstubs.c [hunk 1]
```diff
@@ -346,16 +346,57 @@ NtUserSetSysColors(
    return Ret;
 }
 
-DWORD
+BOOL APIENTRY UserUpdateInputContext(PIMC pIMC, DWORD dwType, DWORD_PTR dwValue)
+{
+    PTHREADINFO pti = GetW32ThreadInfo();
+    PTHREADINFO ptiIMC = UserHMGetHandle(pIMC);
```
Comment (HBelusca) [inline win32ss/user/ntuser/ntstubs.c:352]: This is strange. PIMC is not a handle, and PTHREADINFO is not a handle either.

File: win32ss/user/ntuser/ntstubs.c [hunk 2]
```diff
@@ -346,16 +346,57 @@ NtUserSetSysColors(
...
+    switch (dwType)
+    {
+        case UIC_CLIENTIMCDATA:
+            if (pIMC->dwClientImcData)
+                return FALSE;
+            pIMC->dwClientImcData = dwValue;
+            break;
+
+        case UIC_IMEWINDOW:
+            pIMC->hImeWnd = (HWND)dwValue;
```
Comment (HBelusca) [inline win32ss/user/ntuser/ntstubs.c:366]: Shouldn't there be some sort of validation that the given value is indeed a valid window handle?

File: sdk/include/reactos/undocuser.h [hunk 3]
```diff
@@ -296,6 +296,12 @@ MessageBoxTimeoutW(
 
 LPCWSTR WINAPI MB_GetString(IN UINT wBtn);
 
+/* dwType for NtUserUpdateInputContext */
+enum UPDATE_INPUT_CONTEXT
```
Comment (HBelusca) [inline sdk/include/reactos/undocuser.h:300]: Not `typedef enum _UPDATE_INPUT_CONTEXT { ... } UPDATE_INPUT_CONTEXT;` ?

File: win32ss/user/ntuser/ntstubs.c (hunk 1 above)
Comment (katahiromz) [reply win32ss/user/ntuser/ntstubs.c]: Corrected to `pIMC->head.pti`.

File: win32ss/user/ntuser/ntstubs.c (hunk 2 above)
Comment (katahiromz) [reply win32ss/user/ntuser/ntstubs.c]: Validation added.

File: sdk/include/reactos/undocuser.h (hunk 3 above)
Comment (katahiromz) [reply sdk/include/reactos/undocuser.h]: Fixed.

**Pull request merged**